sql-plugin/src/main/scala/com/nvidia/spark/rapids/aggregate.scala

```diff
@@ -94,6 +94,28 @@ class GpuHashAggregateMeta(
             s"'final', 'complete', or 'all'")
       }
     }
+    hashAggReplaceMode match {
+      case "partial" => {
+        // In partial mode, if there are non distinct functions and multiple distinct functions
+        // non distinct functions are computed using First operator. The final result would be
+        // incorrect for non distinct functions for partition size > 1. Falling back to CPU for
+        // this special case.
```

Comment:
Can we elaborate why having a First fails here?

Reply:
Updated it. Please let me know if it's better now.